sympy/functions/combinatorial/factorials.py

```diff
@@ -235,6 +235,52 @@ def _eval_Mod(self, q):
 
                     return S(fc % q)
 
+    def factmultiplicity(self, p, n):
```

Comment:
What about making multiplicity recognize the unevaluated factorial instead:
```
>>> factorial(10,evaluate=0)
factorial(10)
```
So if multiplicity gets `isinstance(foo, factorial) and foo.args[0].is_Integer` you can call a private method to do the calculation.

Reply:
can you please elaborate...

Reply:
I partially got what you said, and had a thought about it, but how to implement it, I am not able to figure out.

Reply:
Oh. I see. You made it a method of factorial. But calling it looks ackward. I would recommend:
```python
multiplicity(p, factorial(10, evaluate=0)) call a function (not method) `factmultiplicity`
```
Maybe someone else has a better idea. Perhaps, too, it would be nice to have `Factorial` be an unevaluated form of `factorial` in keeping with many other objects/functions in SymPy. That was one would write `Factorial(10)` as a shortcut for `factorial(10, evaluate=False)`.

Reply:
> Oh. I see. You made it a method of factorial. But calling it looks ackward. I would recommend:
> 
> ```python
> multiplicity(p, factorial(10, evaluate=0)) call a function (not method) `factmultiplicity`
> ```
> 
> Maybe someone else has a better idea. Perhaps, too, it would be nice to have `Factorial` be an unevaluated form of `factorial` in keeping with many other objects/functions in SymPy. That was one would write `Factorial(10)` as a shortcut for `factorial(10, evaluate=False)`.

Then is multiplicity a method of factorial??

Reply:
I have added it as a function in factor_.py